api/src/certification/complementary-certification/domain/usecases/attach-badges-to-complementary-certification.js

```diff
@@ -0,0 +1,154 @@
+import { NotFoundError } from '../../../../../lib/domain/errors.js';
+import { DomainTransaction } from '../../../../../lib/infrastructure/DomainTransaction.js';
+import uniq from 'lodash/uniq.js';
+
+import bluebird from 'bluebird';
+import { ComplementaryCertificationBadgeToAttach } from '../models/ComplementaryCertificationBadgeToAttach.js';
+import { InvalidBadgeLevelError } from '../../../shared/domain/errors.js';
+
+const attachBadgesToComplementaryCertification = async function ({
+  complementaryCertificationId,
+  userId,
+  targetProfileIdToDetach,
+  complementaryCertificationBadgesToAttachDTO,
+  complementaryCertificationBadgeToAttachValidator,
+  badgeRepository,
+  complementaryCertificationRepository,
+  complementaryCertificationBadgesRepository,
+}) {
+  _verifyThatLevelsAreConsistent({
+    complementaryCertificationBadgesToAttachDTO,
+  });
+
+  await _verifyThatComplementaryCertificationExists({
+    complementaryCertificationId,
+    complementaryCertificationRepository,
+  });
+
+  await _verifyThatBadgesToAttachExist({
+    complementaryCertificationBadgesToAttachDTO,
+    badgeRepository,
+  });
+
+  const complementaryCertificationBadgesToAttach = complementaryCertificationBadgesToAttachDTO.map(
+    (complementaryCertificationBadgeToAttachDTO) => {
+      const complementaryCertificationBadgeToAttach = ComplementaryCertificationBadgeToAttach.from({
+        ...complementaryCertificationBadgeToAttachDTO,
+        complementaryCertificationId,
+        userId,
+      });
+      complementaryCertificationBadgeToAttachValidator.validate(complementaryCertificationBadgeToAttach);
+      return complementaryCertificationBadgeToAttach;
+    },
+  );
```

Comment:
Proposition pour alléger la lecture 

```suggestion
  const complementaryCertificationBadgesToAttach = complementaryCertificationBadgesToAttachDTO.map(
    _toDomain({ userId, complementaryCertificationId, complementaryCertificationBadgeToAttachValidator }),
  );
  
  ///// plus une fonction privée
  /// Note:  la fonction toDomain pourrait être améliorée, voir mon commentaire sur la validator
  
  
  function _toDomain({ userId, complementaryCertificationId, complementaryCertificationBadgeToAttachValidator }) {
  return (complementaryCertificationBadgeToAttachDTO) => {
    const complementaryCertificationBadgeToAttach = ComplementaryCertificationBadgeToAttach.from({
      ...complementaryCertificationBadgeToAttachDTO,
      complementaryCertificationId,
      userId,
    });
    complementaryCertificationBadgeToAttachValidator.validate(complementaryCertificationBadgeToAttach);
    return complementaryCertificationBadgeToAttach;
  };
}
```